statefun-docs/docs/concepts/logical.rst

```diff
@@ -0,0 +1,79 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+   or more contributor license agreements.  See the NOTICE file
+   distributed with this work for additional information
+   regarding copyright ownership.  The ASF licenses this file
+   to you under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+
+.. _logical-functions:
+
+#################
+Logical Functions
+#################
+
+Stateful Function's are allocated logically, which means the system can support an unbounded number of instances with a finite amount of resources.
+Logical instances do not use CPU, memory, or threads when not actively being invoked, there is no theoretical upper limit on the number of instances that can created.
+Users are encouraged to model their applications as granularly as possible, based on what makes the most sense for their application, instead of desigining applications around resource constraints.
+
+.. contents:: :local:
+
+.. _address:
+
+Function Address
+================
+
+In a local environment, the address of an object is the same as a reference to it.
+But in a Stateful Function's application, function instances are virtual and their runtime location is not exposed to the user.
+Instead, an ``Address`` is used to reference a specific stateful functions in the system..
+
+.. code-block:: proto
+
+  syntax = "proto3";
```

Comment:
Why use protobuf to illustrate this?
It seems unrelated, also might confuse users that they need to construct `Address` as a protobuf message.

Reply:
I didn't want to to be language specific, I will create an illustration.